client/pdo/client/commands/collection.py

```diff
@@ -0,0 +1,197 @@
+        if k == 'save_file' :
+            save_files.append(v)
+        elif isinstance(v, dict):
+            save_files.extend(__find_contracts__(v))
+
+    return save_files
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+@experimental
+def export_contract_collection(context, context_paths : typing.List[str], contract_cache : str, export_file : str) :
+    """Export the context and associated contract files to a zip file that
+    can be shared with others who want to use the contract
+
+    @type context: pdo.client.builder.context.Context
+    @param context: current context
+    @param context_paths : list of path expressions to retrieve values from a context
+    @param contract_cache : name of the directory where contract save files are stored
+    @param export_file : name of the file where the contract family will be written
+    """
+
+    # the context we create is initialized, mark it so
+    export_context = {
+        'contexts' : context_paths,
+        'initialized' : True,
+    }
+
+    # copy the portions of the context specified in the context_paths
+
+    # note: while there are fields in the context that are unnecessary for future use of the
+    # contract, it is far easier to simply copy them here. at some point, this may be smarter about
+    # only copying the fields that are necessary.
+
+    for c in context_paths :
+        # since the incoming contexts are paths, we need to make sure
+        # we copy the context from/to the right location
+        (*prefix, key) = c.split('.')
+        ec = export_context
+        for p in prefix :
+            if p not in ec :
+                ec[p] = {}
+            ec = ec[p]
+        ec[key] = copy.deepcopy(context.get(c))
+
+    # now find all of the contract references in the exported context
+    save_files = __find_contracts__(export_context)
+
+    # and write the contract collection into the zip file
+    with ZipFile(export_file, 'w') as zf :
+        # add the context to the package, this has a canonical name
+        zf.writestr('context.toml', toml.dumps(export_context))
+
+        # add the contract save files to the package
+        for s in save_files :
+            contract_file_name = os.path.join(contract_cache, s)
+            zf.write(contract_file_name, arcname=s)
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+@experimental
+def import_contract_collection(context_file_name : str, contract_cache : str, import_file : str) -> dict :
+    """Import the context and contract files from a collections zip file
+
+    @param context_file_name : name of the file to save imported context
+    @param contract_cache : name of the directory where contract save files are stored
+    @param import_file : name of the contract collection file to import
+    @rtype: dict
```

Comment:
isn't this redundant with the interface return type annotation?

Reply:
yes and no. you can autodoc the file and these will work. again... this was really an experiment with the full (standard) syntax for documentation

Reply:
not fixing this